http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala

```diff
@@ -441,7 +441,14 @@ class AKSInterpreter[F[_]](config: AKSInterpreterConfig,
                                         namespaceName
       )
 
-      // Update the relay listener deployment
+      // The app is blocked in updating now
+      _ <- appQuery.updateStatus(app.id, AppStatus.Updating).transaction
+
+      // Update the relay listener deployment if the app is not already updating from a message submission
+      // TODO: discuss this check. (if (dbApp.app.status != AppStatus.Updating))
+      // TODO: This might be sufficient, but really, we would need to store >>what version<< the app is being updated to determine if we should pass-through to polling or ignore
+      // TODO: the issue is that if we simply exit out if `Updating` status, then the app can remain stuck in updating if leo restarts during polling
+      // TODO: perhaps monitor at boot should just grab all apps in Updating and transition them to running + save an error to avoid this complexity?
```

Comment:
this is a discussion for a separate PR, I just want to leave a large TODO section indicating not to merge this PR until the subsequent ticket is written